tools/cellprofiler/cellprofiler.xml

```diff
@@ -0,0 +1,108 @@
+            <conditional name="input_type">
+                <param name="type" type="select" label="Select either a zip file or a collection from history">
+                    <option value="zip">zip file</option>
+                    <option value="images">images</option>
+                </param>
+                <when value="zip">
+                    <param name="inputdir" type="data" format="zip" label="Select image dataset from history" /> 
+                </when>
+                <when value="images">
+                    <param name="inputdir" type="data" multiple="true" format="tiff,tif" label="Images"/>
+                </when>         
+            </conditional>
+    </inputs>
+
+    <outputs>
+        <collection name="pipeline_output" type="list" label="Cellprofiler pipeline output files">
+            <discover_datasets pattern="__designation_and_ext__" visible="false" directory="output"/>
+        </collection>
+    </outputs>
+
+    <tests>
+        <test>
+            <param name="pipeline" value="ExampleHuman.cppipe" />
+            <conditional name="input_type">
+                <param name="type" value="images"/>
+                <param name="inputdir" value="images/AS_09125_050116030001_D03f00d0.tif,images/AS_09125_050116030001_D03f00d1.tif,images/AS_09125_050116030001_D03f00d2.tif" />
+            </conditional>
+            <output_collection name="pipeline_output" type="list">
+                <element name="AS_09125_050116030001_D03f00d0.tif_Overlay">
+                    <assert_contents>
+                      <has_size value="183808"/>
+                    </assert_contents>
+                </element>
+                <element name="Cells">
+                    <expand macro="test_assert_content" n="290" />
+                </element>
+                <element name="Cytoplasm">
+                  <expand macro="test_assert_content" n="290" />
+                </element>
+                <element name="Experiment">
+                  <expand macro="test_assert_content" n="259" />
+                </element>
+                <element name="Image">
+                  <expand macro="test_assert_content" n="2" />
+                </element>
+                <element name="Nuclei">
+                  <expand macro="test_assert_content" n="290" />
+                </element>
+                <element name="PH3">
+                  <expand macro="test_assert_content" n="21" />
+                </element>
+            </output_collection>
+        </test>
+    </tests>
+    
+    <help>
+    This tool runs a complete Cellprofiler v3.1.9 pipeline file. It takes a pipeline file and images from Galaxy history as input and generates its output files in Galaxy history. The input images can be a single zip file or multiple files.
+
+    *** Parameters ***
+
+    images: single zip file or multi-select image files
+    
+    pipeline: CellProfiler v3.1.9 pipeline file(.cppipe) file.
```

Comment:
get rid of the version number ... everywhere ... maybe some better description ...

Reply:
Not sure about this. people usually know well which version they are using because the features might change...

Reply:
But the tool is versioned and should have the 3.1.9 version. If its really needed here it should be a macro so that with an update you only need to update one place.